lib/finch.ex

```diff
@@ -26,8 +58,33 @@ defmodule Finch do
   ]
   @atom_to_method Enum.zip(@atom_methods, @methods) |> Enum.into(%{})
 
-  def request(method, url, headers, body, opts) do
+  def start_link(opts) do
+    name = Keyword.fetch!(opts, :name)
+    pools = Keyword.get(opts, :pools, %{})
+
+    config = %{
+      registry_name: name,
+      manager_name: manager_name(name),
+      supervisor_name: pool_supervisor_name(name),
+      pools: pools
+    }
+
+    Supervisor.start_link(__MODULE__, config, name: supervisor_name(name))
+  end
+
+  def init(config) do
+    children = [
+      {DynamicSupervisor, name: config.supervisor_name, strategy: :one_for_one},
+      {Registry, [keys: :duplicate, name: config.registry_name, meta: [config: config]]},
+      {PoolManager, config}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
```

Comment:
It should probably be `:one_for_all` (safer) until we can stop and think about the different failure modes and restarts.